File: app/src/main/java/com/example/flightstats/FlightListItem.java
package com.example.flightstats;

/**
 * Enriched flight item for display — combines Room Flight with airport city/country lookups.
 */
public class FlightListItem {
    public int id;
    public String origin;
    public String destination;
    public String date;
    public double distance;
    public String flightNumber;
    public String airline;
    public String seat;
    public String seatClass;
    public String notes;
    public String departureTime;
    public String arrivalTime;

    // Resolved from Airport table
    public String originCity;
    public String originCountry;   // ISO 2-letter e.g. "NL"
    public String destCity;
    public String destCountry;

    public static String countryToFlag(String isoCode) {
        return "";
    }
}

File: app/src/main/java/com/example/flightstats/FlightListItem.kt
package com.example.flightstats

data class FlightListItem(
    val id: Int,
    val origin: String,
    val destination: String,
    val date: String?,
    val distance: Double,
    val flightNumber: String?,
    val airline: String?,
    val seat: String?,
    val seatClass: String?,
    val notes: String?,
    val departureTime: String?,
    val arrivalTime: String?,
    val originCity: String?,
    val originCountry: String?,
    val destCity: String?,
    val destCountry: String?
) {
    companion object {
        fun countryToFlag(isoCode: String?): String {
            if (isoCode == null || isoCode.length != 2) return ""
            val firstChar = Character.codePointAt(isoCode.uppercase(), 0) - 0x41 + 0x1F1E6
            val secondChar = Character.codePointAt(isoCode.uppercase(), 1) - 0x41 + 0x1F1E6
            return String(Character.toChars(firstChar)) + String(Character.toChars(secondChar))
        }
    }
}
